var/spack/repos/builtin/packages/environment-modules/package.py

```diff
@@ -3,64 +3,81 @@
+        '3.2.10', sha256='fb05c82a83477805a1d97737a9f0ca0db23f69b7bce504f1609ba99477b03955',
+        url='http://prdownloads.sourceforge.net/modules/modules-3.2.10.tar.gz'
+    )
 
     variant('X', default=True, description='Build with X functionality')
 
     # Dependencies:
     depends_on('tcl', type=('build', 'link', 'run'))
+    depends_on('tcl@8.4:', type=('build', 'link', 'run'), when='@4.0.0:')
 
-    def install(self, spec, prefix):
-        tcl_spec = spec['tcl']
-
-        # We are looking for tclConfig.sh
-        tcl_config_name = 'tclConfig.sh'
-        tcl_config_dir_options = [tcl_spec.prefix.lib,
-                                  tcl_spec.prefix.lib64]
-
-        tcl_config_found = False
-        for tcl_config_dir in tcl_config_dir_options:
-            tcl_config_found = os.path.exists(
-                join_path(tcl_config_dir, tcl_config_name))
-            if tcl_config_found:
-                break
+    depends_on('py-sphinx', type='build', when='@4.0.0:')
 
-        if not tcl_config_found:
-            raise InstallError('Failed to locate ' + tcl_config_name)
+    def install(self, spec, prefix):
+        tcl = spec['tcl']
 
-        # See: https://sourceforge.net/p/modules/bugs/62/
-        cpp_flags = ['-DUSE_INTERP_ERRORLINE']
+        # Determine where we can find tclConfig.sh
+        for tcl_lib_dir in [tcl.prefix.lib, tcl.prefix.lib64]:
+            tcl_config_file = os.path.join(tcl_lib_dir, 'tclConfig.sh')
+            if os.path.exists(tcl_config_file):
+                break
+        else:
+            raise InstallError('Failed to locate tclConfig.sh')
 
         config_args = [
+            "--prefix=" + prefix,
             "--without-tclx",
             "--with-tclx-ver=0.0",
-            "--prefix=" + prefix,
             # It looks for tclConfig.sh
-            "--with-tcl=" + tcl_config_dir,
-            "--with-tcl-ver=%d.%d" % (
-                tcl_spec.version.version[0], tcl_spec.version.version[1]),
-            '--disable-debug',
+            "--with-tcl=" + tcl_lib_dir,
+            "--with-tcl-ver={0}.{1}".format(*tcl.version.version[0:2]),
             '--disable-dependency-tracking',
             '--disable-silent-rules',
             '--disable-versioning',
-            '--datarootdir=' + prefix.share,
-            'CPPFLAGS=' + ' '.join(cpp_flags)
+            '--datarootdir=' + prefix.share
         ]
 
         if '~X' in spec:
             config_args = ['--without-x'] + config_args
 
+        if '@4.0.0:' in self.spec:
+            config_args.extend([
+                '--with-tclsh={0}'.format(tcl.prefix.bin.tclsh)
+            ])
+
+        if '@3.2.10' in self.spec:
+            # See: https://sourceforge.net/p/modules/bugs/62/
+            config_args.extend([
+                '--disable-debug',
+                'CPPFLAGS=-DUSE_INTERP_ERRORLINE'
+            ])
+
         configure(*config_args)
-        make()
+
+        targets = ['-C', 'doc', 'all'] if '@4.0.0:' else []
```

Comment:
@xdelaruelle Can you please double check the target I am building? Also, would it be fine with you if I add your name as a maintainer of the package :slightly_smiling_face:

Reply:
A bare `make` call should be fine. Why do you add specific targets ?

Fine to get named as maintainer of the package :slightly_smiling_face:

Reply:
> A bare make call should be fine. Why do you add specific targets ?

Following the advice in `README.md`. Do I need then the dependency on Sphinx?

Reply:
I see. README is unclear, I will clarify that. Installation instructions are available in https://github.com/cea-hpc/modules/blob/master/INSTALL.rst. The build is achieved with a basic `./configure; make; make install`.

I suggest to disable the build of the compatibility version (compilation of version 3.2 of Modules in addition of version 4) by adding the `--disable-compat-version` argument to the `./configure` script.

Requirements to build from a source tarball are:
* bash
* make
* sed

Then to run, Tcl >= 8.4 is needed.

Sphinx is not needed when building from a source tarball (only needed when building from the git repository)